torchtune/data/_messages.py

```diff
@@ -682,17 +707,38 @@ class AlpacaToMessages(Transform):
     due to this custom logic.
 
     Args:
-        train_on_input (bool): Whether the model is trained on the user prompt or not.
-            Default is True.
+        masking_strategy (Optional[str]): masking strategy to use for model training.
+            Must be one of: `train_on_all`, `train_on_assistant`, `train_on_last`.
+            Default is "train_on_all".
+            - ``train_on_all``: both user and assistant messages are unmasked
+            - ``train_on_assistant``: user messages are masked, only assistant messages are unmasked
+            - ``train_on_last``: only the last assistant message is unmasked
         column_map (Optional[Dict[str, str]]): a mapping to change the expected "instruction", "input",
             and "output" column names to the actual column names in the dataset. Default is None,
             keeping the default column names.
+        train_on_input (Optional[bool]): whether the model is trained on the user prompt or not.
+            Deprecated parameter and will be removed in a future release.
+            Default is None.
     """
 
     def __init__(
-        self, train_on_input: bool = True, column_map: Optional[Dict[str, str]] = None
+        self,
+        masking_strategy: Optional[str] = "train_on_all",
```

Comment:
contrary to the other comment, this default value should remain `train_on_all` since `train_on_input` was defaulted to True. so let's keep this